Display supported versions in UnsupportedFormat error

kornelski opened the pull request:
I've got this error, and wasn't sure if my rustdoc was too new or too old, so I've added more info to the error.

obi1kenobi commented:
Ah, this is a good idea but not the right way to do it.

Just about everything in `src/` is code-generated from a template based on the supported versions. So instead you'll want to:
- Edit the template: https://github.com/obi1kenobi/trustfall-rustdoc/tree/main/template
- Run [`scripts/regenerate_templated_code.sh`](https://github.com/obi1kenobi/trustfall-rustdoc/blob/main/scripts/regenerate_templated_code.sh) to regenerate the code from the template.
- Inspect whether you got what you wanted out or not, then either check in both the codegen and template changes, or iterate. There's a CI job that ensures the codegen and template match.

obi1kenobi commented:
> I've got this error, and wasn't sure if my rustdoc was too new or too old

Also, as a reminder, the cargo-semver-checks release notes always list the supported Rust versions. The most-recently released version supports Rust 1.84+, and the upcoming version will be 1.85+: https://github.com/obi1kenobi/cargo-semver-checks/releases/tag/v0.41.0

kornelski commented:
I've updated the template, and added a warning about the autogeneration gotcha.

I appreciate well written release notes, but being able to diagnose problems without having to check the release notes is even better :)

obi1kenobi commented on src/versioned.rs:
```diff
@@ -269,3 +271,18 @@ impl<'a> VersionedRustdocAdapter<'a> {
+        43,
+        #[cfg(feature = "v45")]
+        45,
+        #[cfg(feature = "v48")]
+        48,
```
```suggestion
        #[cfg(feature = "v53")]
        53,
```
